tests/test_toolchain.py

```diff
@@ -0,0 +1,62 @@
+import sys
+import pytest
+import mock
+from pythonforandroid.toolchain import ToolchainCL
+from pythonforandroid.util import BuildInterruptingException
+
+
+class TestToolchainCL:
+
+    def test_help(self):
+        """
+        Calling with `--help` should print help and exit 0.
+        """
+        argv = ['toolchain.py', '--help', '--storage-dir=/tmp']
+        with mock.patch('sys.argv', argv), pytest.raises(SystemExit) as ex_info, \
+                mock.patch('argparse.ArgumentParser.print_help') as m_print_help:
+            ToolchainCL()
+        assert ex_info.value.code == 0
+        assert m_print_help.call_args_list == [mock.call()]
+
+    @pytest.mark.skipif(sys.version_info < (3, 0), reason="requires python3")
+    def test_unknown(self):
+        """
+        Calling with unknown args should print help and exit 1.
+        """
+        argv = ['toolchain.py', '--unknown']
+        with mock.patch('sys.argv', argv), pytest.raises(SystemExit) as ex_info, \
+                mock.patch('argparse.ArgumentParser.print_help') as m_print_help:
+            ToolchainCL()
+        assert ex_info.value.code == 1
+        assert m_print_help.call_args_list == [mock.call()]
+
+    def test_create(self):
+        """
+        Basic `create` distribution test.
+        """
+        argv = ['toolchain.py', 'create', '--sdk-dir=/tmp/android-sdk', '--ndk-dir=/tmp/android-ndk']
+        with mock.patch('sys.argv', argv), \
```

Comment:
Again here, the backslashes in this context manager:
```
        with mock.patch('sys.argv', argv), mock.patch(
            'pythonforandroid.build.get_available_apis'
        ) as m_get_available_apis, mock.patch(
            'pythonforandroid.build.get_toolchain_versions'
        ) as m_get_toolchain_versions, mock.patch(
            'pythonforandroid.build.get_ndk_platform_dir'
        ) as m_get_ndk_platform_dir, mock.patch(
            'pythonforandroid.build.get_cython_path'
        ) as m_get_cython_path, mock.patch(
            'pythonforandroid.toolchain.build_dist_from_args'
        ) as m_build_dist_from_args:
            m_get_available_apis.return_value = [27]
            m_get_toolchain_versions.return_value = (['4.9'], True)
            m_get_ndk_platform_dir.return_value = (
                '/tmp/android-ndk/platforms/android-21/arch-arm',
                True,
            )
```
This will also be pep8 friendly

Reply:
I also hate backslashes, but for context managers, I haven't found a more readable way to write it so far.
I'm not yet convinced this way is more readable, but I gave it a try nonetheless :smile: